javalib/src/main/scala/java/io/Reader.scala

```diff
@@ -53,8 +55,20 @@ abstract class Reader private[this] (_lock: Option[Object])
   def skip(n: Long): Long = {
     if (n < 0)
       throw new IllegalArgumentException("Cannot skip negative amount")
-    else if (read() == -1) 0
-    else 1
+    else {
+      @tailrec
+      def loop(m: Long, lastSkipped: Long): Long = {
+        if (m <= 0)
+          lastSkipped
+        else {
+          val mMin = Math.min(m, 8192).toInt
+          val skipped = Math.max(read(new Array[Char](mMin), 0, mMin), 0)
```

Comment:
Consider allocating this array only once, outside of the loop, for example with
```scala
val buffer = new Array[Char](Math.min(n, 8192).toInt)
```

Reply:
Also, I think the new implementation fails to stop the loop if the input runs into EOF before `n` chars have been skipped. You'll need to stop the loop if `read` returns `< 0` (and return `lastSkipped` in that case).

(gosh, looks like this method is much trickier than it looks at first)